File: src/daybreak/abilitywar/game/games/mode/GameManifest.java
package daybreak.abilitywar.game.games.mode;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface GameManifest {

	String Name();
	
	String[] Description();
	
}

File: modules/Plugin/src/daybreak/abilitywar/game/GameManifest.kt
package daybreak.abilitywar.game

import daybreak.abilitywar.utils.library.MaterialX
import org.bukkit.Material
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.CLASS

@Retention(RUNTIME)
@Target(CLASS)
annotation class GameManifest(val name: String, val description: Array<String>)

@Retention(RUNTIME)
@Target(CLASS)
annotation class Category(val value: GameCategory) {
	enum class GameCategory(val icon: Material, val displayName: String) {
		GAME(Material.DIAMOND_SWORD, "게임"),
		MINIGAME(Material.FEATHER, "미니 게임"),
		DEBUG(MaterialX.COMMAND_BLOCK.parseMaterial(), "디버그");
	}
}

annotation class GameAliases(vararg val value: String)
